script.js

```diff
@@ -0,0 +1,155 @@
+
+  dateInfo.innerText = `${year}.${month + 1}.${date} ${
+    dayList[day]
+  } ${hours}시`;
+};
+
+//submit이벤트 핸들러
+form.addEventListener('submit', function (event) {
+  // 기본 제출 동작을 막고 할일 추가함수 실행
+  event.preventDefault();
+  inputTodo();
+});
+
+//할일을 입력받아 todoList에 추가 후 리렌더링
+const inputTodo = () => {
+  //입력값 유효성 검사
+  if (input.value.trim() === '') {
+    return;
+  }
+  //배열추가
+  todoList.push(input.value);
+  localStorage.setItem('todos', JSON.stringify(todoList));
+  input.value = '';
+
+  //리렌더링
+  renderTodo();
+};
+
+//로컬스토리지 내 todo 목록 띄우기
+const renderTodo = () => {
+  todoList = JSON.parse(localStorage.getItem('todos')) || [];
+  todoNum.innerText = todoList.length;
+  todoContent.innerHTML = ''; //html 초기화
+  todoList.forEach((todo, index) => {
+    //요소 생성 후 띄우기
+    const li = document.createElement('li');
+
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.addEventListener('change', () => {
+      addDone(todo);
+      removeTodo(index);
+    });
+
+    const span = document.createElement('span');
+    span.innerText = todo;
+
+    const deleteBtn = document.createElement('button');
+    deleteBtn.innerText = 'X';
+    deleteBtn.addEventListener('click', () => removeTodo(index));
+
+    // li 요소에 체크박스, 제목, 삭제 버튼을 추가
+    li.appendChild(checkbox);
+    li.appendChild(span);
+    li.appendChild(deleteBtn);
+
+    // todoContent에 li 요소를 추가
+    todoContent.appendChild(li);
+  });
+};
+
+//로컬스토리지 내 done 목록 띄우기
+const renderDone = () => {
+  doneList = JSON.parse(localStorage.getItem('dones')) || [];
+  doneNum.innerText = doneList.length;
+  doneContent.innerHTML = ''; //html 초기화
+  doneList.forEach((done, index) => {
+    //요소 생성 후 띄우기
+    const li = document.createElement('li');
+
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.addEventListener('change', () => {
+      addTodo(done);
+      removeDone(index);
+    });
+
+    const span = document.createElement('span');
+    span.innerText = done;
+
+    const deleteBtn = document.createElement('button');
+    deleteBtn.innerText = 'X';
+    deleteBtn.addEventListener('click', () => removeDone(index));
+
+    // li 요소에 체크박스, 제목, 삭제 버튼을 추가한다
+    li.appendChild(checkbox);
+    li.appendChild(span);
+    li.appendChild(deleteBtn);
+
+    // doneContent에 li 요소를 추가한다
+    doneContent.appendChild(li);
+  });
+};
+
+//todoList삭제함수
+const removeTodo = (index) => {
+  //filter메소드로 삭제한 배열로 업데이트
+  todoList = todoList.filter((data) => {
+    return data !== todoList[index];
+  });
+  //localStorage다시 저장
+  localStorage.setItem('todos', JSON.stringify(todoList));
+
+  renderTodo(); //리렌더링
+};
+
+//doneList삭제함수
+const removeDone = (index) => {
+  //filter메소드로 삭제한 배열로 업데이트
+  doneList = doneList.filter((data) => {
+    return data !== doneList[index];
+  });
+  //localStorage다시 저장
+  localStorage.setItem('dones', JSON.stringify(doneList));
+  renderDone(); //doneList 리렌더링
+};
```

Comment:
이 부분 로직이 겹치는 부분이 많은 것 같은데, 'todos' 혹은 'dones'을 매개변수로 받아서 아래와 같이 함수를 합칠수도있을것같네요
```suggestion
const removeFunc = (query, index) => {
  if(query === 'todos'){
  [targetList, targetFunc] = todoList, renderTodo;
  } else if (query === 'dones'){
    [targetList, targetFunc] = doneList, renderDone;
  }
  targetList = targetList.filter((data) => {
    return data !== targetList[index];
  });
  localStorage.setItem(query, JSON.stringify(targetList));
  targetFunc();

```